src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1382,15 +1382,183 @@ void C2_MacroAssembler::string_indexof_linearscan(Register haystack, Register ne
   bind(DONE);
 }
 
+// Compare longwords
+void C2_MacroAssembler::string_compare_long_same_encoding(Register result, Register str1, Register str2,
+                                                  const bool isLL, Register cnt1, Register cnt2,
+                                                  Register tmp1, Register tmp2, Register tmp3,
+                                                  const int STUB_THRESHOLD, Label *STUB, Label *SHORT_STRING, Label *DONE) {
+  Label TAIL_CHECK, TAIL, NEXT_WORD, DIFFERENCE;
+
+  const int base_offset = isLL ? arrayOopDesc::base_offset_in_bytes(T_BYTE)
+                               : arrayOopDesc::base_offset_in_bytes(T_CHAR);
```

Comment:
I don't think we need to distinuish `T_BYTE` and `T_CHAR` here. The reason is that the character storage used for both Latin1 and UTF16 strings is always a byte array [1]. So we should always use `T_BYTE` for both cases.

[1] https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/String.java#L160

Reply:
Thanks.
As mentioned in this PR's description, there will be another follow-up pr later, I can do it in that one.